src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.h

```diff
@@ -41,9 +42,32 @@ class DefaultDACVerifier : public DeviceAttestationVerifier
                                                    const ByteSpan & attestationSignatureBuffer,
                                                    const Crypto::P256PublicKey & dacPublicKey, const ByteSpan & csrNonce) override;
 
-protected:
     DefaultDACVerifier() {}
 
+protected:
+    AttestationVerificationResult CheckDacPaiVidPids(const DeviceAttestationVerifier::AttestationInfo & info,
+                                                     Crypto::AttestationCertVidPid & dacVidPid,
+                                                     Crypto::AttestationCertVidPid & paiVidPid);
+
+    AttestationVerificationResult CheckAttestationSignature(const DeviceAttestationVerifier::AttestationInfo & info);
+
+    AttestationVerificationResult CheckPAA(const DeviceAttestationVerifier::AttestationInfo & info,
+                                           DeviceInfoForAttestation & deviceInfo, Platform::ScopedMemoryBuffer<uint8_t> & paaCert,
+                                           MutableByteSpan & paaDerBuffer, Crypto::AttestationCertVidPid & paaVidPid,
+                                           Crypto::AttestationCertVidPid & paiVidPid);
+
+    AttestationVerificationResult CheckCertTimes(const DeviceAttestationVerifier::AttestationInfo & info,
+                                                 MutableByteSpan & paaDerBuffer);
+
+    AttestationVerificationResult CheckCertChain(const DeviceAttestationVerifier::AttestationInfo & info,
+                                                 MutableByteSpan & paaDerBuffer);
+
+    AttestationVerificationResult CheckCertDeclaration(const DeviceAttestationVerifier::AttestationInfo & info,
+                                                       MutableByteSpan & paaDerBuffer, Crypto::AttestationCertVidPid & dacVidPid,
+                                                       Crypto::AttestationCertVidPid & paiVidPid,
+                                                       Crypto::AttestationCertVidPid & paaVidPid,
+                                                       DeviceInfoForAttestation & deviceInfo);
+
```

Comment:
Please document these methods, they are definitely no obvious and some of the names are misleading.

Overall the construction here is much harder to follow and it's much more difficult to audit the code for validity against the device attestation procedure.

Already all the code here in existing `VerifyAttestationInformation` is very small and contains only "building-block" calls.

I suggest the following approach:

- Bring back the existing structure
- Rename CloudDeviceAttestationVerifier to `DacOnlyPartialAttestationVerifier` and just do the checks that don't relate to the  *trust chain* part of attestation, just the "structural" portion that can be done locally.

Then you just include the steps you want from existing DefaultDeviceAttestationVerifier.

This would have many advantages:

- Easier to audit/maintain
- Does not pretend by name to verify things with a cloud or otherwise have anything to do with the cloud
- Allows you to do only the parts of the attestation procedure that can be done locally without access to a PAA trust store.
- Does not change significant structural elements.

Reply:
Reverted this approach